src/main/java/de/tum/in/www1/artemis/service/competency/CompetencyRelationService.java

```diff
@@ -57,27 +58,30 @@ public CompetencyRelation getCompetencyRelation(Competency tailCompetency, Compe
      *
      * @param tailCompetency the tail Competency
      * @param headCompetency the head Competency
-     * @param type           the type of the relation
+     * @param relationType   the type of the relation
      * @param course         the course the relation belongs to
      * @return the persisted CompetencyRelation
      */
-    public CompetencyRelation createCompetencyRelation(Competency tailCompetency, Competency headCompetency, String type, Course course) {
-        RelationType relationType;
-        try {
-            relationType = RelationType.valueOf(type);
-        }
-        catch (IllegalArgumentException e) {
-            throw new BadRequestException("Invalid value for relation type");
-        }
-
+    public CompetencyRelation createCompetencyRelation(Competency tailCompetency, Competency headCompetency, RelationType relationType, Course course) {
         var relation = getCompetencyRelation(tailCompetency, headCompetency, relationType);
         var competencies = competencyRepository.findAllForCourse(course.getId());
         var competencyRelations = competencyRelationRepository.findAllWithHeadAndTailByCourseId(course.getId());
         competencyRelations.add(relation);
+
         if (competencyService.doesCreateCircularRelation(competencies, competencyRelations)) {
             throw new BadRequestException("You can't define circular dependencies between competencies");
         }
 
         return competencyRelationRepository.save(relation);
     }
+
+    /**
+     * Converts a CompetencyRelation to a CompetencyRelationDTO
+     *
+     * @param relation the relation
+     * @return the CompetencyRelationDTO
+     */
+    public CompetencyRelationDTO relationToDTO(CompetencyRelation relation) {
+        return new CompetencyRelationDTO(relation.getId(), relation.getTailCompetency().getId(), relation.getHeadCompetency().getId(), relation.getType());
```

Comment:
Could we move this method to the DTO? (`CompetencyRelationDTO.of(CompetencyRelation relation)`)